### PR TITLE
kubelet: Refactor prober.

### PR DESCRIPTION
**Pull request by yifan-gu**
Decompose the health check prober from the kubelet.

@yujuhong @dchen1107 @vmarmol

### CODE REVIEW

File: pkg/kubelet/probe.go [hunk 1]
```diff
@@ -35,50 +37,83 @@ import (
...
+	runner dockertools.ContainerCommandRunner
+
+	readinessManager *kubecontainer.ReadinessManager
+	refManager       *kubecontainer.RefManager
+	recorder         record.EventRecorder
+}
+
+// NewProber creates a Prober, it takes a command runner and
+// several container info managers.
+func NewProber(
```
Comment (yujuhong) [inline pkg/kubelet/probe.go]: Do we expect the function to be called outside of this package? Why capitalized the name?

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: @yujuhong Yes, I expecting to move it to kubelet/prober, otherwise container runtime cannot import it.

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: Or move it to kubelet/container...

Actually this function does belong to kubelet in some sense... Hopefully if kubelet someday does not import dockertools or rkt directly, we can move them back...

Comment (yifan-gu) [inline pkg/kubelet/probe.go]: Another way to work around this is to define an interface in kubelet/container:
``` go
type prober interface {
    Probe()
}
```

And then we can pass the kubelet as this interface to the runtime without the circular import problem.

But I feel it's cleaner to just move it to kubelet/prober. What's your opinion?

**Comment by yujuhong** [comment]
I am fine with it being in kubelet/prober :)

**Pull request merged**